ci/azure-pipelines-nightly.yml

```diff
@@ -108,8 +108,8 @@ jobs:
     displayName: "Run all ROM E2E tests"
   - template: ../ci/publish-bazel-test-results.yml
 
-- job: slow_otbn_crypto_tests
-  displayName: "Slow OTBN Crypto Tests"
+- job: nightly_tagged_sw_tests
+  displayName: "Software tests explicitly tagged as 'nightly'"
```

Comment:
I'm not sure about this name change. It's a job in the nightly pipeline, so it's kind of "nightly" already! I think the tag on the test means "Only run this test in the nightly run" and the name here means "The stuff that wouldn't be run otherwise".

I don't have strong feelings, but I'm not sure this name change is needed.